src/odin_devices/dac63004.py

```diff
@@ -0,0 +1,392 @@
+        """specify a current range for the dac at the provided index 
+        
+        current_range (string):
+        "0000": 0 μA to 25 μA,
+        "0001": 0 μA to 50 μA,
+        "0010": 0 μA to 125 μA,
+        "0011": 0 μA to 250 μA,
+        "0100": 0 μA to -24 μA,
+        "0101": 0 μA to -48 μA,
+        "0110": 0 μA to -120 μA,
+        "0111": 0 μA to -240 μA,
+        "1000": -25 μA to +25 μA,
+        "1001": -50 μA to +50 μA,
+        "1010": -125 μA to +125 μA,
+        "1011": -250 μA to +250 μA,
+        index (int): the index of the dac we want to specify the current range for
+        """
+        #Change the current range for the DAC with the index provided. The value provided should be a binary string, one of those listed above. 
+        #See page 58 for more info on the DAC_X_IOUT_MISC_CONFIG register
+        self.read_modify_write("DAC_" + str(index) + "_IOUT_MISC_CONFIG", 0b0001111000000000, int(current_range + "000000000", 2))
+        
+    def set_dac_voltage(self, index, voltage, referenceVoltage):
+        """Set the output for the given index to the given voltage based on the reference voltage
+
+        Args:
+            index (int): the index of the dac we want to specify the voltage for
+            voltage (int): the voltage we want to set the output to
+            referenceVoltage (int): the reference voltage being used by the dac
+        """
+        value_to_write = voltage*4096/referenceVoltage 
+        self.set_dac_output(str(bin(value_to_write)).replace("0b", ""), index, False)
+    
+    def set_dac_current_micro_amps(self, index, current):
+        """ Set the current range to an approriate value for the current requested, then calculate the value that needs to be written to get the current we want and write that value.
+
+        Args:
+            index (int): the index of the dac we want to set the current for
+            current (int): the amount of microamps we want to set the current to
+        """
+        if (current > 250 or current < -240):
+            raise Exception("Invalid current - current must be between -240 and 250.")
+        range = "0000"
+        #get the range either based on the current entered
+        set_current = "0000"
+        if (current > 125):
+            set_current = "0011"
+        elif (current > 50):
+            set_current = "0010"
+        elif (current > 25):
+            set_current = "0001"
+        elif (current > 0):
+            set_current = "0000"
+        elif (current > -24):
+            set_current = "0100"
+        elif (current > -48):
+            set_current = "0101"
+        elif (current > -120):
+            set_current = "0110"
+        elif (current > -240):
+            set_current = "0111"
+        elif (current > -250):
+            set_current = "1011"
+        self.write_register(self.read_register_address("DAC_" + str(index) + "_IOUT_MISC_CONFIG"), int(set_current + "000000000", 2), True)
+        range = set_current
```

Comment:
It's a little confusing here what the distinction between current and range is here